### ISSUE TITLE
Unable to update image preview if layer's position and size or pixels are locked

### ISSUE THREAD

**Issue by strixaluco** [khalim19/gimp-plugin-export-layers#74]
#### What went wrong?

```
Traceback (most recent call last):
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/gui/gui_preview_image.py", line 396, in _get_image_preview
    keep_image_copy=True)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 232, in export
    self._export_layers()
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 443, in _export_layers
    self._process_and_export_item(layer_elem)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 453, in _process_and_export_item
    layer_copy = self._process_layer(layer_elem, self._image_copy, layer)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 517, in _process_layer
    ["after_insert_layer"], [image, layer_copy, self], additional_args_position=0)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/pygimplib/pgoperations.py", line 263, in execute
    _execute_executor(item.operation, group)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/pygimplib/pgoperations.py", line 245, in _execute_executor
    [group], additional_args, additional_kwargs, additional_args_position)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/pygimplib/pgoperations.py", line 261, in execute
    _execute_operation(operation, operation_args, operation_kwargs)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/pygimplib/pgoperations.py", line 207, in _execute_operation
    return operation(*args, **kwargs)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/gui/gui_preview_image.py", line 427, in _scale_layer_for_layer_exporter
    (layer.offsets[1] + layer.height) * self._preview_scaling_factor)
RuntimeError: Item 'background copy' (1012) cannot be modified because its position and size are locked
```

#### If you can, tell us the steps to reproduce the problem.
1. Select any given layer
2. Click `Lock pixels` or `Lock position and size`
3. Launch `Export layers`

#### Configuration

Version of Export Layers: 3.3.1

Version of GIMP: 2.10.8

Operating system: Debian GNU/Linux 9 (stretch) 64 bit

**Comment by khalim19** [comment]
Thanks, this is definitely a bug. I presume this only happens in the preview?

In the meantime, you can disable the preview to avoid spamming the error dialog (click on the arrow and uncheck `Update preview automatically`).

**Comment by strixaluco** [comment]
Okay, now there is another issue while trying to actually export locked layers, not preview them:
```
RuntimeError: The layer to merge down to is locked.
```
Steps to reproduce are basically the same as in original post:

1. Select any given layer
2. Click `Lock pixels` or `Lock position and size`
3. Launch `Export layers`
4. Include locked layer in the export
5. Hit `Export`

Full traceback below:

```
Traceback (most recent call last):
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/gui/gui_plugin.py", line 198, in func_wrapper
    func(self, *args, **kwargs)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/gui/gui_plugin.py", line 821, in _on_button_export_clicked
    self._layer_exporter.export()
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 232, in export
    self._export_layers()
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 443, in _export_layers
    self._process_and_export_item(layer_elem)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 453, in _process_and_export_item
    layer_copy = self._process_layer(layer_elem, self._image_copy, layer)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 522, in _process_layer
    additional_args_position=0)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/pygimplib/pgoperations.py", line 259, in execute
    operation, operation_args, operation_kwargs, group)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/pygimplib/pgoperations.py", line 227, in _execute_operation_with_foreach_operations
    operation, operation_args, operation_kwargs)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/pygimplib/pgoperations.py", line 207, in _execute_operation
    return operation(*args, **kwargs)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 776, in _execute_operation
    return operation(*operation_args, **operation_kwargs)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/exportlayers.py", line 768, in _operation
    function(image, layer, layer_exporter, *new_args, **new_kwargs)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/builtin_procedures.py", line 58, in insert_foreground_layer
    _insert_tagged_layer(image, layer_exporter, tag, position=0)
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/builtin_procedures.py", line 96, in _insert_tagged_layer
    _insert_merged_tagged_layer(image, layer_exporter, tag, position))
  File "/home/strix/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/export_layers/builtin_procedures.py", line 128, in _insert_merged_tagged_layer
    image, image.layers[i], gimpenums.EXPAND_AS_NECESSARY)
RuntimeError: The layer to merge down to is locked.
```

**Comment by khalim19** [comment]
Fixed in 3.3.2.
